src/models/create_token_transaction.ts

```diff
@@ -242,6 +251,71 @@ class CreateTokenTransaction extends Transaction {
 
     return tx;
   }
+
+  /**
+   * Checks if this transaction is the creation of an NFT following the NFT Standard Creation.
+   * @see https://github.com/HathorNetwork/rfcs/blob/master/text/0032-nft-standard.md#transaction-standard
+   * @throws {NftValidationError} Will throw an error if the NFT is not valid
+   *
+   * @param {Network} network Network to get output addresses first byte
+   * @returns {void} If this function does not throw, the NFT is valid
+   */
+  validateNftCreation(network: Network): void {
+    // An invalid transaction will fail here too
+    this.validate();
+
+    // No need to check the tx version, it is enforced by the class constructor
+
+    /*
+     * NFT creation must have at least a DataScript output (the first one) and a Token P2PKH output.
+     * Also validating maximum outputs of transactions in general
+     */
+    if (this.outputs.length < 2) {
+      throw new NftValidationError(`Tx has less than the minimum required amount of outputs`);
+    }
+
+    // Validating the first output
+    const firstOutput = this.outputs[0];
+
+    // NFT creation DataScript output must have value 1 and must be of HTR
+    if (firstOutput.value !== 1 || !firstOutput.isTokenHTR()) {
+      throw new NftValidationError(`First output is not a valid NFT data output`);
+    }
+    // NFT creation Datascript must be of type data
+    if (!(firstOutput.parseScript(network) instanceof ScriptData)) {
+      throw new NftValidationError(`First output is not a DataScript`);
+    }
+
+    // Iterating on all but the first output for validation and counting authorities
+    let mintOutputs = 0;
+    let meltOutputs = 0;
+    for (let index=1; index < this.outputs.length; ++index) {
+      const output = this.outputs[index];
+
+      // Must have a valid length
+      if (!output.hasValidLength()) {
+        throw new InvalidOutputsError(`Output at index ${index} script is too long.`);
+      }
+
+      // Ensuring the type of the output is valid
+      const validTypes = [OutputType.P2PKH.toString(), OutputType.P2SH.toString()];
+      const outputType = output.getType(network)?.toLowerCase() || '';
+      if (!validTypes.includes(outputType)) {
+        throw new NftValidationError(`Output at index ${index} is not of a valid type`);
+      }
+
+      // Counting authority outputs
+      if (output.isAuthority()) {
+        mintOutputs += (output.value & TOKEN_MINT_MASK) > 0 ? 1 : 0;
+        meltOutputs += (output.value & TOKEN_MELT_MASK) > 0 ? 1 : 0;
+      }
```

Comment:
```suggestion
      mintOutputs += output.isMint() ? 1 : 0;
      meltOutputs += output.isMelt() ? 1 : 0;
```
The `isMint` and `isMelt` already check for `isAuthority` and the mint/melt masks.

Reply:
Done on [f8fe02f](https://github.com/HathorNetwork/hathor-wallet-lib/pull/378/commits/f8fe02f650ad84931363f9c636791fc61c504d85).